### DISCUSSION TITLE
Rauchwarnmelder HmIP-SWSD

### DISCUSSION THREAD

**Discussion by ChristophCaina** [SukramJ/aiohomematic#759] [Ideas]
At the moment, the HmIP provides a sensor with the alter-states "idle_off", "primary_alarm", "secondary_alarm" and "intrusion_alarm".

with upcomming changes in HomeAssistant, it could probably changed to an "alert" entity, which would allow to make use of the following color-options:

frontent PR: https://github.com/home-assistant/core/pull/83926

To provide the information of the alarm type, this could maybe set as an attribute or similar - the important information at least, is the alarm status (alarm on / off / idle)

**Comment by SukramJ** [comment]
Please add a link to the corresponding alert developer documentation.

**Reply by SukramJ** [reply]
The biggest question is: What is the concept behind an alert?

Checking the dev source code i can see service like `turn_on` and `turn_on`. Looks like the alerts stay for a longer time.
The SWSD report states while they are alarming and end when you press the device buttion. This is more a short time event.
How could this match?

If you have a good and concrete suggestion that would be helpful.

**Reply by ChristophCaina** [reply]
thanks for the information :)
I was just going through the current PRs on HomeAssistant to see, what might be come in future releases (especially, after they have started to develop further color-options for entities ;) )

I've checked the Documentation for the "alert" integration...
It seems like a Watcher for other entities to send notifications and provide "alarm information" if a specific condition was met...

Here's a link to the general documentation for "alert"
https://www.home-assistant.io/integrations/alert

--> at least, an alert can be configured to perform notification and Alert-Tasks for (probably) any entity that is available.

Anyway - I've just had a short look how the Homematic IP Integration from HA does handle the different Alarms from the SWSD, since I knew from the time when I used the HmIP Integration, that it was created as a Binary Sensor...

In this case, the binary Sensor will be ON independend of the exact alarm type.
The Alarm Type information will then be provided as an Attribute to the Sensor
```
@property
    def extra_state_attributes(self) -> dict[str, Any]:
        """Return the state attributes of the security group."""
        state_attr = super().extra_state_attributes

        smoke_detector_at = getattr(self._device, "smokeDetectorAlarmType", None)
        if smoke_detector_at:
            if smoke_detector_at == SmokeDetectorAlarmType.PRIMARY_ALARM:
                state_attr[ATTR_SMOKE_DETECTOR_ALARM] = str(smoke_detector_at)
            if smoke_detector_at == SmokeDetectorAlarmType.INTRUSION_ALARM:
                state_attr[ATTR_INTRUSION_ALARM] = str(smoke_detector_at)
        return state_attr
```

My preferred method would either be to provide the alarm status as a binary sensor (alarm on / alarm off => idle)
and then provide the exact state / alarm type information within an attribute of the sensor - or within another entity "alarm_type"
I know, that it would be possible to implement that with an template - but I prefer "out of the box" options when possible ... :)

For now, I will go with the template option and try to create binary sensors that would only be on / off - and use the state information as an attribute... but maybe it can be considered for future releases?

**Reply by ChristophCaina** [reply]
Another Idea would be to create at least another entity for the "intrusion_alarm" - if the swsd will be used as an alarm device for intrusion alarms...
That would allow to configure different cases directly:

1. binary_sensor: smoke detecor (catches primary + secondary alarm)
2. binary_sensor: intrusion alarm (catches only intrusion_alarm)

**Reply by SukramJ** [reply]
This sensor is auto generated and contains all required information for further actions based on its state.

```
      "smoke_detector_alarm_status": {
        "state": {
          "idle_off": "Untätig",
          "primary_alarm": "Primär-Alarm",
          "intrusion_alarm": "Einbruch-Alarm",
          "secondary_alarm": "Sekundär-Alarm"
        }
```
Using attributes is a step backwards in my opinion as it requires two states to be evaluated.

If you want a binary_sensor --> create a template
if you want a notification --> use it in an automation / create a blueprint

If you want to support the community -> The Wiki is open and you could create a page with templates to support other users.
You could also package more complex solutions within a blueprint that we publish here to support others.
Like: create a notification/alert if SWSD Wohnzimmer switches state to secondary_alarm. This could be manually dismissed by the user later.

Currently i don't see/understand a use case for alerts (entity integration) within this integration.

Apart from a technical solution my opinion regarding HmIP-SWSD:
This device has fixed battery that that should last for 10 years. Used as a smoke detector and for nothing else, otherwise the batteries could be empty within month.
Buy a much HmIP-ASIR, that is cheaper, louder and has replaceable batteries and does not has to save your ass when it is burning.

**Reply by ChristophCaina** [reply]
> Apart from a technical solution my opinion regarding HmIP-SWSD:
This device has fixed battery that that should last for 10 years. Used as a smoke detector and for nothing else, otherwise the batteries could be empty within month.
Buy a much HmIP-ASIR, that is cheaper, louder and has replaceable batteries and does not has to save your ass when it is burning.

That's clear - and personally, I don't use the SWSD as an intrusion alarm system... It's only being used as smoke detector.

btw:
shouldn't be the "sabotagekontakt" (other devices) be device type: tamper ?

**Reply by SukramJ** [reply]
> shouldn't be the "sabotagekontakt" (other devices) be device type: tamper ?

Sounds good

**Comment by ChristophCaina** [comment]
here's my template for getting the "Icon Coloring" and split the Alarm Type...

```
#Smoke Detector Alarm
  - binary_sensor:
      - name: "Rauchwarnmelder Arbeitszimmer"
        unique_id: "rauchwarnmelder_arbeitszimmer"
        icon: >
          {% if states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['idle_off', 'intrusion_alarm'] %}
            mdi:smoke-detector-variant
          {% elif states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['primary_alarm','secondary_alarm'] %}
            mdi:smoke-detector-variant-alert
          {% endif %}
        device_class: smoke
        state: >
          {% if states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['idle_off', 'intrusion_alarm'] %}
            off
          {% elif states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['primary_alarm', 'secondary_alarm'] %}
            on
          {% endif %}
        availability: >
          {{ states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') not in ['unknown', 'unavailable', 'None'] }}
        attributes:
          Device State: >
            {% if is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'idle_off') %}
              Untätig
            {% elif is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'primary_alarm') %}
              Primärer Alarm - Rauch erkannt!
            {% elif is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'secondary_alarm') %}
              Sekundärer Alarm - durch anderen Rauchmelder ausgelöst
            {% endif %}

# Intrusion Alarm Device
  - binary_sensor:
      - name: "Einbruchsalarm Arbeitszimmer"
        unique_id: "intrusion_alarm_arbeitszimmer"
        icon: >
          {% if states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['idle_off', 'secondary_alarm', 'primary_alarm'] %}
            mdi:shield
          {% elif is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'intrusion_alarm') %}
            mdi:shield-alert
          {% endif %}
        device_class: tamper
        state: >
          {% if states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') in ['idle_off', 'secondary_alarm, 'primary_alarm'] %}
            off
          {% elif is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'intrusion_alarm') %}
            on
          {% endif %}
        availability: >
          {{ states('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status') not in ['unknown', 'unavailable', 'None'] }}
        attributes:
          Device State: >
            {% if is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'idle_off') %}
              Untätig
            {% elif is_state('sensor.rauchwarnmelder_arbeitszimmer_smoke_detector_alarm_status', 'intrusion_alarm') %}
              Einbruch erkannt!
            {% endif %}
```
Status: Normal
![grafik](https://user-images.githubusercontent.com/26391061/207884551-3f04b768-c832-493c-826c-0677de6269a6.png)
![grafik](https://user-images.githubusercontent.com/26391061/207884513-b066f62e-5781-4ad0-97e0-ec3f3c6f519c.png)

Status: Alarm - Primär-Alarm (Rauch erkannt)
![grafik](https://user-images.githubusercontent.com/26391061/207885138-35299d6e-b229-4dd5-b582-af6685ae1219.png)
![grafik](https://user-images.githubusercontent.com/26391061/207884945-20234cb9-8961-4858-ab47-b292b601884a.png)

Status: Alarm - Sekundär-Alarm (Durch andere Melder ausgelöst)
![grafik](https://user-images.githubusercontent.com/26391061/207885236-7e8cc4c7-c609-4e43-847d-b6269a576b97.png)
![grafik](https://user-images.githubusercontent.com/26391061/207885256-8a7960ad-f320-43dc-925e-300ee29a0904.png)

Status: Alarm - Intrusion-Alarm (Einbruchsalarm)
![grafik](https://user-images.githubusercontent.com/26391061/207887293-64a88553-4827-4d8b-9e10-0a3d84c5718f.png)
![grafik](https://user-images.githubusercontent.com/26391061/207889538-9f4ad25c-4f8b-4745-8d14-bff3fd1701c9.png)

**Reply by ChristophCaina** [reply]
in addition:

Unavailable:
![grafik](https://user-images.githubusercontent.com/26391061/207890006-cf39b4cd-6acc-4650-937c-f0a7fab0ecf1.png)

Unknown:
![grafik](https://user-images.githubusercontent.com/26391061/207890028-5ab32c38-84d8-4a3d-9f66-bd3df9d18c90.png)

**Reply by SukramJ** [reply]
There is a [subpage](https://github.com/danielperna84/hahomematic/wiki/Community-Templates-for-Home-Assistant) in the wiki as starting point. Please add them directly.
If you need/want more pages then feel free to add/update them by yourself.

**Comment by Diggen85** [comment]
Thanks for all the Information and also for the Template but if we look at the HmIP-SWD (Waterlevel Senor), it also has 3 binary_sensors - moisture_detected, waterlevel_detected, alarmstate
I think it would be nice if we have the same Sensor/Entity-Formating for the same HmIP-DeviceClass (HmIP-SWD / HmIP-SWSD)

**Reply by SukramJ** [reply]
> I think it would be nice if we have the same Sensor/Entity-Formating for the same HmIP-DeviceClass (HmIP-SWD / HmIP-SWSD)

If you want same semantics you should use templates.

Both devices are generated (see docs), and in case of the SWSD we only provide state translations.
HmIP-SWD has three parameters `MOISTURE_DETECTED`, `WATERLEVEL_DETECTED` and `ALARM_STATE` which are of type BOOLEAN. 
HmIP-SWSD has one parameter  `SMOKE_DETECTOR_ALARM_STATUS` that is of type ENUM. 
So both devices are designed different by eQ-3.
